audit_history/tests/test_audit_history.py

```diff
@@ -16,25 +16,25 @@ def test_empty_history_after_save(self):
 
     def test_history_after_save_with_audit(self):
         self.blog_post.save_with_audit_record(None, self.history_event, payload=self.payload)
-        self.assertEqual(self.payload[1], self.blog_post.history[0].get('payload').get(1))
+        self.assertEqual(self.payload['a'], self.blog_post.history[0]['payload']['a'])
 
     def test_save_with_audit_record_with_staff_user(self):
         self.blog_post.save_with_audit_record(self.user, self.history_event, payload=self.payload)
         self.assertEqual(self.user_name, self.blog_post.history[0]['actor']['name'])
         self.assertEqual(self.user_email, self.blog_post.history[0]['actor']['email'])
         self.assertEqual(True, self.blog_post.history[0]['actor']['is_staff'])
-        self.assertEqual(self.payload[1], self.blog_post.history[0]['payload'][1])
-        self.assertEqual(self.payload[2], self.blog_post.history[0]['payload'][2])
+        self.assertEqual(self.payload['a'], self.blog_post.history[0]['payload']['a'])
+        self.assertEqual(self.payload['b'], self.blog_post.history[0]['payload']['b'])
 
     def test_save_with_audit_record_without_user(self):
         self.blog_post.save_with_audit_record(None, self.history_event, payload=self.payload)
         self.assertEqual(None, self.blog_post.history[0]['actor'])
-        self.assertEqual(self.payload[1], self.blog_post.history[0]['payload'][1])
-        self.assertEqual(self.payload[2], self.blog_post.history[0]['payload'][2])
+        self.assertEqual(self.payload['a'], self.blog_post.history[0]['payload']['a'])
+        self.assertEqual(self.payload['b'], self.blog_post.history[0]['payload']['b'])
 
     def test_n_usages_of_save_with_audit(self):
         count = randint(1, 100)
```

Comment:
Wouldn't it be better if this test was more deterministic and always test e.g. 100 cycles? What's the reason for the random cycle count? (Just trying to understand, I'm probably missing something here.)

Reply:
The main idea is check that count of saving == count of history items

Reply:
But why randomize and not fix to e.g. 100?

Reply:
Updated. Please check https://github.com/nexto/django-model-audit-history/pull/4/commits/81d95f683b6a3806f7f37ec3972898af526e87be